tests/test_upgrade.rb

```diff
@@ -13,88 +13,94 @@
 # limitations under the License.
 
 require_relative 'ciscotest'
-require_relative '../lib/cisco_node_utils/service'
+require_relative '../lib/cisco_node_utils/upgrade'
 
 include Cisco
 
-# TestService - Minitest for feature Service node Utility class
-class TestService < CiscoTestCase
-  @skip_unless_supported = 'service'
+# TestUpgrade - Minitest for router Upgrade node Utility class
+class TestUpgrade < CiscoTestCase
+  @skip_unless_supported = 'upgrade'
 
-  def preconfig_service_info
-    path = File.expand_path('../service_info.yaml', __FILE__)
-    skip('Cannot find tests/service_info.yaml') unless File.file?(path)
+  def preconfig_upgrade_info
+    path = File.expand_path('../upgrade_info.yaml', __FILE__)
+    skip('Cannot find tests/upgrade_info.yaml') unless File.file?(path)
     info = YAML.load(File.read(path))
     valid_info?(info)
     info
   end
 
   def valid_info?(info)
-    skip('tests/service_info.yaml file is empty') unless info
-    msg = 'Missing key in tests/service_info.yaml'
+    skip('tests/upgrade_info.yaml file is empty') unless info
+    msg = 'Missing key in tests/upgrade_info.yaml'
     %w(install_image install_media).each do |key|
       skip("#{msg}: #{key}") if info[key].nil?
     end
   end
 
   ###################
-  # Service tests   #
+  # Upgrade tests   #
   ###################
 
   def test_clear_status
-    Service.clear_status
-    refute(Service.upgraded?)
+    Upgrade.clear_status
+    refute(Upgrade.upgraded?)
   end
 
   def test_delete
     shell_command('touch /bootflash/foobar')
```

Comment:
Just realized that `shell_command` will create a file on bootflash if we are running this in a `native` or `guestshell` environment but will not work for oac.  The `shell_command` api is just a wrapper around  `run bash` or `run guestshell` but there is no equivalent `run oac` command today.  Not sure if this will work, but see if you can replace `shell_command('touch /bootflash/foobar')` with `config("show version > bootflash:foo.txt")` instead.  You may need to escape the `>`.